homeassistant/components/esphome/select.py

```diff
@@ -140,33 +160,57 @@ async def async_added_to_hass(self) -> None:
 
     async def async_select_option(self, option: str) -> None:
         """Select an option."""
-        if wake_word_id := self._wake_words.get(option):
-            # _attr_current_option will be updated on
-            # async_satellite_config_updated after the device sets the wake
-            # word.
-            self._entry_data.async_assist_satellite_set_wake_word(wake_word_id)
+        self._attr_current_option = option
+        self.async_write_ha_state()
+
+        wake_word_id = self._wake_words.get(option)
+        self._entry_data.async_assist_satellite_set_wake_word(
+            self._wake_word_index, wake_word_id
+        )
 
     def async_satellite_config_updated(
         self, config: AssistSatelliteConfiguration
     ) -> None:
         """Update options with available wake words."""
         if (not config.available_wake_words) or (config.max_active_wake_words < 1):
-            self._attr_current_option = None
+            # No wake words
             self._wake_words.clear()
+            self._attr_current_option = NO_WAKE_WORD
+            self._attr_options = [NO_WAKE_WORD]
             self.async_write_ha_state()
             return
 
         self._wake_words = {w.wake_word: w.id for w in config.available_wake_words}
-        self._attr_options = sorted(self._wake_words)
+        self._attr_options = [NO_WAKE_WORD, *sorted(self._wake_words)]
 
+        option = NO_WAKE_WORD
         if config.active_wake_words:
-            # Select first active wake word
-            wake_word_id = config.active_wake_words[0]
-            for wake_word in config.available_wake_words:
-                if wake_word.id == wake_word_id:
-                    self._attr_current_option = wake_word.wake_word
-        else:
-            # Select first available wake word
-            self._attr_current_option = config.available_wake_words[0].wake_word
-
+            for active_wake_word_id in config.active_wake_words:
+                for wake_word, wake_word_id in self._wake_words.items():
+                    if (wake_word_id == active_wake_word_id) and (
+                        self._attr_current_option == wake_word
+                    ):
+                        option = wake_word
+                        break
+
+        self._attr_current_option = option
         self.async_write_ha_state()
+
+
+class EsphomeSecondaryAssistSatelliteWakeWordSelect(
+    EsphomeAssistSatelliteWakeWordSelect
+):
+    """Secondary wake word selector for esphome devices."""
+
+    entity_description = SelectEntityDescription(
+        key="wake_word_2",
+        translation_key="wake_word_2",
+        entity_category=EntityCategory.CONFIG,
+    )
+
+    def __init__(self, entry_data: RuntimeEntryData) -> None:
+        """Initialize a secondary wake word selector."""
+        EsphomeAssistSatelliteWakeWordSelect.__init__(self, entry_data, 1)
```

Comment:
can we just pass the index in the constructor to this class when we initialize it, so we don't have to create a 2nd class.

Index can be used to generate unique ID, entity description key, translation key etc.